www/jury/statistics.php

```diff
@@ -74,37 +85,202 @@
+		.orient("left");
+
+	svg.append("g")
+		.attr("class", "x axis")
+		.attr("transform", "translate(0," + height + ")")
+		.call(xAxis);
+
+	svg.append("g")
+		.attr("class", "y axis")
+		.attr("transform", "translate(0,0)")
+		.call(yAxis);
+
+	var layer = svg.selectAll(".layer")
+		.data(layers)
+		.enter().append("g")
+		.attr("class", "layer")
+		.style("fill", function(d, i) {
+			return charts[i].color;
+		})
+		.style("fill-opacity", function(d, i) {
+			return charts[i].bars.fill;
+		}).attr("clip-path", "url(#clip)");
+
+	var gap = 0.1;
+	var rangeGap = x(<?= $bar_size ?>) - x(0);
+	var rect = layer.selectAll("rect")
+		.data(function(d) {
+			return d;
+		})
+		.enter().append("rect")
+		.attr("x", function(d) {
+			return x(d.x+gap*<?= $bar_size ?>);
+		})
+		.attr("y", function(d) {
+			return y(d.y0 + d.y);
+		})
+		.attr("width", (1-2*gap)*rangeGap)
+		.attr("height", function(d) {
+			return y(d.y0) - y(d.y0 + d.y);
 		});
+
+	svg.append("rect")
+		.attr("x", 0)
+		.attr("y", 0)
+		.attr("height", height)
+		.attr("width", width)
+		.style("stroke", "black")
+		.style("fill", "none")
+		.style("stroke-width", 1);
+
+	var textstyle = "13px 'DejaVu Sans',Arial";
+
+	svg.selectAll(".axis")
+		.style("fill", "none")
+		.style("stroke", "black")
+		.style("stroke-opacity", 0.15);
+	svg.selectAll("text")
+		.style("font", textstyle)
+		.style("stroke", "none")
+		.style("fill", "black");
+	var caption = d3.select("#stats_graph")
+		.append("g")
+		.attr("transform", "translate(" + (margin.left + 15) + "," + (margin.top + 8) + ")")
+	var captionBackground = caption.append("rect")
+
+	hdist = 18; 
+	var off = 0;
+	charts.forEach(function(c) {
+		off += hdist;
+		var gap = 0.35;
+		caption.append("rect")
+			.attr("y",off-hdist + hdist*gap)
+			.attr("x",0)
+			.attr("width", 15) 
+			.attr("height", hdist*(1-gap))
+			.style("fill", c.color);
+		caption.append("text")
+			.text(c.label)
+			.attr("y",off)
+			.attr("x",20)
+			.style("font", textstyle)
+	}); 
+
+	var bbox = caption.node().getBBox();
+	var captionMargin = 7;
+	captionBackground
+		.attr("width", bbox.width + 2*captionMargin)
+		.attr("height", bbox.height + 2*captionMargin)
+		.attr("x", bbox.x - captionMargin)
+		.attr("y", bbox.y - captionMargin)
+		.style("fill", "rgb(255,255,255)")
+		.style("fill-opacity", 0.7);
+
+	function utf8_to_b64(str) {
+		return window.btoa(unescape(encodeURIComponent(str)));
+	}
+
+	placeholder.append("br");
+	placeholder.append("a")
+		.attr("download", "<?=isset($shortname) ? $shortname . "_" : ""?>stats.svg")
+		.attr("href", "data:image/svg+xml;base64," +
+			utf8_to_b64("<svg xmlns=\"http://www.w3.org/2000/svg\" width='" + $("#placeholder").width() +
+			"' height='" + $("#placeholder").height()+ "'>" +
+			new XMLSerializer().serializeToString(document.getElementById('stats_graph')) + "</svg>"))
+		.text("download graph");
+	placeholder.append("p")
+		.text("If the above link does not work for you, please try another browser.");
```

Comment:
Perhaps helpful to state what browser would suffice/in which browsers it works? Do we even support browsers where it doesn't?

Reply:
I don't remember exactly what I tested. Will have to retest and clarify.

Reply:
I did some quick testing (on up to date browsers only):
- Internet Explorer does not download anything at all.
- (desktop) Safari navigates to the image instead of downloading.
- Chrome and Firefox look good.